test/functional/api/v2/hostgroups_controller_test.rb

```diff
@@ -63,6 +63,32 @@ class HostgroupsControllerTest < ActionController::TestCase
         assert_response :success
         assert_equal assigns('hostgroup').ansible_roles, [@ansible_role2, @ansible_role1]
       end
+
+      test 'should append a role to a hostgroup with a correct ordering' do
+        hostgroup = FactoryBot.create(:hostgroup,
+                                      :ansible_role_ids => [@ansible_role1.id])
+        post :ansible_roles,
+             :params => {
+               :id => hostgroup.id,
+               :ansible_role_id => [@ansible_role2.id]
+             },
+             :session => set_session_user
+        assert_response :success
+        assert_equal assigns('hostgroup').ansible_roles, [@ansible_role1, @ansible_role2]
+      end
+
+      test 'should remove only specified roles from a hostgroup' do
+        hostgroup = FactoryBot.create(:hostgroup,
+                                      :ansible_role_ids => [@ansible_role2.id, @ansible_role1.id])
+        delete :ansible_roles,
+               :params => {
+                 :id => hostgroup.id,
+                 :ansible_role_id => [@ansible_role2.id]
+               },
+               :session => set_session_user
+        assert_response :success
```

Comment:
HTTP 204 (No Content) is common after a DELETE. If it's 200 Success, I wonder what's in the content.

Reply:
We have the same behaviour with responses almost everywhere else since we use the same logic/methods. In this case the server will respond with `200` and the modified entity (hostgroup in this case).